modin/core/execution/ray/implementations/pandas_on_ray/partitioning/virtual_partition.py

```diff
@@ -279,26 +298,39 @@ def apply(
             In this case, we have to return the partitioning to its previous
             orientation (the lengths will remain the same). This is ignored between
             two axis partitions.
+        lengths : list, optional
+            The list of lengths to shuffle the object.
+        manual_partition : bool, default: False
+            If True, partition the result with `lengths`.
         **kwargs : dict
             Additional keywords arguments to be passed in `func`.
 
         Returns
         -------
         list
             A list of `PandasOnRayDataframeVirtualPartition` objects.
+
+        Notes
+        -----
+        The ``args`` iterable is deserialized before being passed to the superclass ``apply``
+        method. In older versions of Modin, ``args`` was instead passed internally as
+        ``kwargs["args"]``, and this deserialization was handled in a special case in
+        ``deploy_ray_func``, making control flow difficult to follow.
         """
         if not self.full_axis:
             # If this is not a full axis partition, it already contains a subset of
             # the full axis, so we shouldn't split the result further.
             num_splits = 1
         if len(self.call_queue) > 0:
             self.drain_call_queue()
-        kwargs["args"] = args
         result = super(PandasOnRayDataframeVirtualPartition, self).apply(
             func,
-            num_splits,
-            other_axis_partition,
-            maintain_partitioning,
+            *deserialize(args),
```

Comment:
This might be one of the places where we get slow as we are blocked in the main thread by calling ray.get. We likely want to call ray.get in worker processes in order for us not to be blocked. Maybe we should be thinking of a more graceful handling of the parameters.

Reply:
Yes, that makes sense. Since this issue seems to be in a virtual partition `apply` call, I've managed to find a mini-benchmark on AWS which exhibits a slowdown, so I'll continue looking into this.